src/sql/src/plan/statement/acl.rs

```diff
@@ -536,14 +546,117 @@ pub fn describe_alter_default_privileges(
     _: &StatementContext,
     _: AlterDefaultPrivilegesStatement<Aug>,
 ) -> Result<StatementDesc, PlanError> {
-    bail_unsupported!("ALTER DEFAULT PRIVILEGES")
+    Ok(StatementDesc::new(None))
 }
 
 pub fn plan_alter_default_privileges(
-    _: &StatementContext,
-    _: AlterDefaultPrivilegesStatement<Aug>,
+    scx: &StatementContext,
+    AlterDefaultPrivilegesStatement {
+        target_roles,
+        target_objects,
+        grant_or_revoke,
+    }: AlterDefaultPrivilegesStatement<Aug>,
 ) -> Result<Plan, PlanError> {
-    bail_unsupported!("ALTER DEFAULT PRIVILEGES")
+    let object_type = (*grant_or_revoke.object_type()).into();
+    match object_type {
+        ObjectType::View | ObjectType::MaterializedView | ObjectType::Source => sql_bail!(
+            "{object_type}S is not valid for ALTER DEFAULT PRIVILEGES, use TABLES instead"
+        ),
+        ObjectType::Sink | ObjectType::ClusterReplica | ObjectType::Role | ObjectType::Func => {
+            sql_bail!("{object_type}S do not have privileges")
+        }
```

Comment:
Could test these

Reply:
This is actually unreachable because it's caught in the parser. I just thought this would be a bit more defensive than `unreachable!()`.